### PR TITLE
add exceptionsList for Codeception

### PR DESCRIPTION
**Pull request by wirwolf**
https://codeception.com/docs/04-FunctionalTests

### CODE REVIEW

**Comment by wirwolf** [comment]
```
Failures
 - kb2999226 (exited 1058) - Error while running 'C:\ProgramData\chocolatey\lib\KB2999226\tools\chocolateyinstall.ps1'.
 See log for details.
 - kb2919442 (exited 1058) - Error while running 'C:\ProgramData\chocolatey\lib\KB2919442\tools\ChocolateyInstall.ps1'.
 See log for details.
 - kb3035131 (exited 1058) - Error while running 'C:\ProgramData\chocolatey\lib\KB3035131\Tools\ChocolateyInstall.ps1'.
 See log for details.
Command exited with code 1058
```
appveyor/pr

**Comment by wirwolf** [comment]
@manuelpichler

File: src/main/php/PHPMD/Rule/Controversial/CamelCaseMethodName.php [hunk 1]
```diff
@@ -45,6 +45,7 @@ class CamelCaseMethodName extends AbstractRule implements MethodAware
         '__set_state',
         '__clone',
         '__debugInfo',
+        '_fixtures'
```
Comment (kylekatarnls) [inline src/main/php/PHPMD/Rule/Controversial/CamelCaseMethodName.php]: As far as I know, `_fixtures` is not a PHP native magic method, so I would say it's up to the user to add it manually on need.

File: src/main/php/PHPMD/Rule/Controversial/CamelCaseVariableName.php [hunk 2]
```diff
@@ -70,8 +70,27 @@ public function apply(AbstractNode $node)
...
+     * @return array
+     */
+    private function getExceptionsList()
+    {
+        try {
+            $exceptions = $this->getStringProperty('exceptions');
+        } catch (\OutOfBoundsException $e) {
+            $exceptions = '';
+        }
+        return explode(',', $exceptions);
```
Comment (kylekatarnls) [inline src/main/php/PHPMD/Rule/Controversial/CamelCaseVariableName.php]: Could we avoid getting `['']` with an exception as an empty string and rather get `[]` when the string is empty.

File: src/main/php/PHPMD/Rule/Controversial/CamelCaseMethodName.php (hunk 1 above)
Comment (ravage84) [reply src/main/php/PHPMD/Rule/Controversial/CamelCaseMethodName.php]: I agree with @kylekatarnls  on this. Please remove this line.

File: src/main/php/PHPMD/Rule/Controversial/CamelCaseVariableName.php (hunk 2 above)
Comment (wirwolf) [reply src/main/php/PHPMD/Rule/Controversial/CamelCaseVariableName.php]: Can not understand idea

**Comment by ravage84** [comment]
@wirwolf please do not merge from other branches or tags such as ``2.6.1`` rebase from ``master`` instead.
This results in a cleaner history and focused PRs.

**Comment by ravage84** [comment]
@wirwolf could you squash all these commits (including mine) into one or two? This is one of the situations where I think the commit history profits from a cleanup through a rebase with squash.

**Comment by kylekatarnls** [comment]
No need for a manual squash, we can use GitHub to Squash the PR rather than merge it.

**Comment by ravage84** [comment]
> 
> 
> @ravage84 No need for a manual squash, we can use GitHub to Squash the PR rather than merge it.

Doesn't that change the owner of the commit?

**Comment by kylekatarnls** [comment]
It changes the commiter, not the author.

**Comment by MarkVaughn** [comment]
lgtm, just add documentation

**Comment by jakzal** [comment]
I know I'm late to the party, but is it possible to cover this feature with tests?

**Comment by ravage84** [comment]
@wirwolf are you still interested in pursuing this?

**Pull request closed without merging**